core/src/main/java/edu/pku/code2graph/model/URI.java

```diff
@@ -13,7 +18,14 @@ public URI() {
     this.lang = "";
     this.file = "";
     this.identifier = "";
-    this.inline = new URI();
+    this.inline = null;
+  }
+
+  public URI(Protocol protocol, String lang, String file, String identifier) {
```

Comment:
since now URI is not loaded directly from config, I will change lang to Language later